File: POO - Python & Java/practica-1/src/gestorAplicacion/Quirofano.java
package gestorAplicacion;

import java.io.Serializable;

public class Quirofano implements Cuarto, Serializable {
	
	/*	INSTANCE VARIABLES	*/
	private static final long serialVersionUID = 1L;
	private Paciente paciente;
	private Doctor doctor;
	
	/*	CONSTRUCTORS	*/
	public Quirofano() {}
	
	public Quirofano(Paciente paciente, Doctor doctor) {
		this.paciente = paciente;
		this.doctor = doctor;
	}
	
	/*	INSTANCE METHODS	*/
	
	/*
	 * despejar: Actualiza el paciente y el doctor a null
	 */
	public void despejar() {
		this.setPaciente(null);
		this.setDoctor(null);
	}
	
	
	/*
	 * apartar: actualiza los atributos de instancia al doctor y pacinete que se pasen como paremetros
	 * @ params: Doctor, Paciente
	 */

	@Override
	public void apartar(Doctor doctor, Paciente paciente) {
		this.setDoctor(doctor);
		this.setPaciente(paciente);
	}
	
	/*
	 * toString: 
	 */
	public String toString() {
		String output;
		if(this.getDoctor() == null) {
			output = "Quirofano vacio";
		} else {
			output = "Quirofano con el doctor " + this.getDoctor().getNombre() + " que esta operando al paciente " + this.getPaciente().getNombre();
		}
		return output;
	}
	
	/*	SETTERS & GETTERS	*/
	@Override
	public Paciente getPaciente() { return this.paciente; }
	@Override
	public void setPaciente(Paciente paciente) { this.paciente = paciente; }
	
	public Doctor getDoctor() { return this.doctor; }
	public void setDoctor(Doctor doctor) { this.doctor = doctor; }
}

File: POO - Python & Java/practica-2/gestorAplicacion/Quirofano.py

from gestorAplicacion.Cuarto import Cuarto

class Quirofano(Cuarto):

    def __init__(self, paciente, doctor):

        self._paciente = paciente
        self._doctor = doctor

    #	
    #	 * despejar: Actualiza el paciente y el doctor a null
    #	 
    def despejar(self):
        self.setPaciente(None)
        self.setDoctor(None)


    #	
    #	 * apartar: actualiza los atributos de instancia al doctor y pacinete que se pasen como paremetros
    #	 * @ params: Doctor, Paciente
    #	 

    def apartar(self, doctor, paciente):
        self.setDoctor(doctor)
        self.setPaciente(paciente)

    #	
    #	 * toString: 
    #	 
    def toString(self):
        output = None
        if self.getDoctor() is None:
            output = "Quirofano vacio"
        else:
            output = "Quirofano con el doctor " + self.getDoctor().getNombre() + " que esta operando al paciente " + self.getPaciente().getNombre()
        return output

    #	SETTERS & GETTERS	
    def getPaciente(self):
        return self._paciente
    def setPaciente(self, paciente):
        self._paciente = paciente

    def getDoctor(self):
        return self._doctor
    def setDoctor(self, doctor):
        self._doctor = doctor
